File: hashen_bitproject_1800135_new/server/src/main/java/bit/project/server/entity/Itemrecive.java
package bit.project.server.entity;

import lombok.Data;
import java.util.List;
import javax.persistence.*;
import java.time.LocalDate;
import javax.persistence.Id;
import javax.persistence.Lob;
import java.time.LocalDateTime;
import lombok.NoArgsConstructor;
import javax.persistence.Entity;
import javax.persistence.GeneratedValue;
import javax.persistence.GenerationType;
import com.fasterxml.jackson.annotation.JsonInclude;
import com.fasterxml.jackson.annotation.JsonIgnoreProperties;

@Data
@Entity
@NoArgsConstructor
@JsonInclude(JsonInclude.Include.NON_NULL)
@JsonIgnoreProperties({"hibernateLazyInitializer", "handler"})
public class Itemrecive{
    @Id
    @GeneratedValue(strategy=GenerationType.IDENTITY)
    private Integer id;

    private String code;

    private LocalDate date;

    @Lob
    private String description;

    private LocalDateTime tocreation;


    @ManyToOne
    private Iorder iorder;

    @OneToMany(mappedBy="itemrecive", cascade = CascadeType.ALL, orphanRemoval = true)
    private List<Itemreciveitem> itemreciveitemList;

    @ManyToOne
    @JsonIgnoreProperties({"creator","status","tocreation","roleList"})
    private User creator;


    public Itemrecive(Integer id){
        this.id = id;
    }

    public Itemrecive(Integer id, String code){
        this.id = id;
        this.code = code;
    }

}

File: hashen_bitproject_1800135_new/client/src/app/entities/itemrecive.ts
import {User} from './user';
import {Iorder} from './iorder';
import {DataPage} from '../shared/data-page';
import {Itemreciveitem} from './itemreciveitem';

export class Itemrecive {
  id: number;
  code: string;
  iorder: Iorder;
  date: string;
  itemreciveitemList: Itemreciveitem[];
  description: string;
  creator: User;
  tocreation: string;
}

export class ItemreciveDataPage extends DataPage{
    content: Itemrecive[];
}
